dbt/include/impala/macros/insertoverwrite.sql

```diff
@@ -45,6 +46,21 @@
         insert overwrite {{ target }} ({{ dest_cols_csv_exclude }}) partition({{ partition_col }})
             select {{ dest_cols_csv }}
             from {{ source }}
+    {% elif partition_cols is not none and raw_strategy == 'append' %}
```

Comment:
Since much of code is similar, Cant we just have it as
if partition_cols is not none
    {% if partition_cols is string %}
           {%- set partition_col = partition_cols -%}
        {% else %}
           {%- set partition_col = partition_cols[0] -%}
        {% endif %}
        {%- set dest_cols_csv = get_quoted_csv_exclude(dest_columns | map(attribute="name"), "") -%}
        {%- set dest_cols_csv_exclude = get_quoted_csv_exclude(dest_columns | map(attribute="name"), partition_col) -%}
    ..
   if raw_strategy == 'insert_overwrite'
        insert overwrite {{ target }} ({{ dest_cols_csv_exclude }}) partition({{ partition_col }})
            select {{ dest_cols_csv }}
            from {{ source }}
   else
        insert into {{ target }} ({{ dest_cols_csv_exclude }}) partition({{ partition_col }})
        (
            select {{ dest_cols_csv }}
            from {{ source }}
        )

Reply:
As discussed, we will take this up as part of 1.2.0 migration

Reply:
postponing this refactoring comment to next release